src/DynamoCore/Models/PresetsModel.cs

```diff
@@ -0,0 +1,190 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Dynamo.Core;
+using Dynamo.Selection;
+using Dynamo.Nodes;
+using System.Xml;
+using Dynamo.Interfaces;
+
+namespace Dynamo.Models
+{
+    /// <summary>
+    /// a class that holds a set of preset design options states
+    /// there is one instance of this class per workspacemodel
+    /// </summary>
+    public class PresetsModel:ILogSource
+    {
+        #region private members
+        private readonly List<PresetState> designStates;
+
+        private void LoadStateFromXml(string name, string description, List<NodeModel> nodes, List<XmlElement> serializednodes, Guid id)
+        {
+            var loadedState = new PresetState(name, description, nodes, serializednodes, id);
+            designStates.Add(loadedState);
+        }
+        #endregion
+
+        # region properties
+        public IEnumerable<PresetState> DesignStates { get { return designStates; } }
+        #endregion
+
+        #region constructor
+        public PresetsModel()
+        {
+            designStates = new List<PresetState>();
+        }
+        #endregion
+
+        #region serialization / deserialzation
+        // we serialze the presets to xml like a model, but we deserialze them before the workspacemodel is constructed
+        //during save and load of the graph, can just inject this into workspacemodel 
+        //when a new designstate is created we'll serialize all the current nodes into a new xmlelement
+        //but not actually write this xml to a file until the graph is saved
+
+        //grabbed some methods needed from modelbase for serialization
+        protected virtual XmlElement CreateElement(XmlDocument xmlDocument, SaveContext context)
```

Comment:
For these comments, let's use the standard `///` commenting style.  Otherwise these comments will not be introduced into the xml docs.